services/blockstorage/service.go

```diff
@@ -197,35 +173,48 @@ func (s *service) loadResultsBlockHeader(height primitives.BlockHeight) (*servic
 
 func (s *service) GetResultsBlockHeader(ctx context.Context, input *services.GetResultsBlockHeaderInput) (result *services.GetResultsBlockHeaderOutput, err error) {
 	err = s.persistence.GetBlockTracker().WaitForBlock(ctx, input.BlockHeight)
-
 	if err == nil {
 		return s.loadResultsBlockHeader(input.BlockHeight)
 	}
-
 	return nil, err
 }
 
-func (s *service) createEmptyTransactionReceiptResult() *services.GetTransactionReceiptOutput {
+func (s *service) createEmptyTransactionReceiptResult(ctx context.Context) (*services.GetTransactionReceiptOutput, error) {
+	out, err := s.GetLastCommittedBlockHeight(ctx, &services.GetLastCommittedBlockHeightInput{})
+	if err != nil {
+		return nil, err
+	}
 	return &services.GetTransactionReceiptOutput{
 		TransactionReceipt: nil,
-		BlockHeight:        s.LastCommittedBlockHeight(),
-		BlockTimestamp:     s.lastCommittedBlockTimestamp(),
-	}
+		BlockHeight:        out.LastCommittedBlockHeight,
+		BlockTimestamp:     out.LastCommittedBlockTimestamp,
+	}, nil
 }
 
+// TODO: are we sure that if we don't find the receipt this API should fail? it should succeed just return nil receipt
```

Comment:
this is a stub implementation, it needs to be rewritten as the spec describes the process using wrong assumptions - i am ignoring these changes as i will rewrite it this month